src/components/base/BaseButton.vue

```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
+import { computed, defineProps } from "vue";
+
+const props = defineProps({
+  buttonType: {
+    type: String,
+    default: "primary",
+  },
+  buttonSize: {
+    type: String,
+    default: "reg",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+const emit = defineEmits(["click"]);
+
+const buttonClasses = computed(() => {
+  return [
+    "button",
+    { "button--primary": props.buttonType === "primary" },
+    { "button--primary-rev": props.buttonType === "primary-rev" },
+    { "button--secondary": props.buttonType === "secondary" },
+    { "button--tertiary": props.buttonType === "tertiary" },
+    { "button--tertiary-rev": props.buttonType === "tertiary-rev" },
+    { "button--sml": props.buttonSize === "sml" },
+    { "button--reg": props.buttonSize === "reg" },
+    { "button--med": props.buttonSize === "med" },
+    { "button--lrg": props.buttonSize === "lrg" },
```

Comment:
Is there any particular need to shorten these instead of going for straight up words like `small`, `regular` (this could be named `default` and be, well the default size for example)`, `medium` etc? 🤔
Especially when you are using Typescript - your IDE would help to autocomplete these values anyway.

Also, because I see you are using typescript here and composition API - you will *love* this section of the docs to make those props more TypeScript friendly and give us much more information 👉  https://vuejs.org/guide/typescript/composition-api.html

Reply:
Actually the link you gave me here along with script setup documentation is what made me fall in love with Vue 3! I dare to say I still have long way to go but I will be on this page a lot.

I am trying to be more verbose when it comes to naming just old habits are hard to beat out :(